src/main-src.js

```diff
@@ -4,45 +4,51 @@ import showList from './showList.js';
 import updateStatus from './updateStatus.js';
 import updateStorage from './updateStorage.js';
 import getStorage from './getStorage.js';
+import addTask from './addTask.js';
+import removeTask from './removeTask.js';
+import removeCompletedTasks from './removeCompletedTasks.js';
+import editTask from './editTask.js';
 
-let tasks = [
-  {
-    index: 0,
-    description: 'First task',
-    completed: false,
-  },
-  {
-    index: 1,
-    description: 'Second task',
-    completed: false,
-  },
-  {
-    index: 2,
-    description: 'Third task',
-    completed: false,
-  },
-];
-
-if (getStorage().length === 0) updateStorage(tasks);
-else tasks = getStorage();
+let tasks = getStorage();
 
 showList(tasks);
 
+// update the task status, edit && remove a task --------------------------------------------------
+
 const listDiv = document.getElementById('list');
-listDiv.addEventListener('change', (event) => {
+const clearAllButton = document.getElementById('clear-list');
+listDiv.addEventListener('click', (event) => {
   if (event.target !== event.currentTarget) {
-    updateStatus(tasks, parseInt(event.target.parentElement.parentElement.id, 10), true);
-    showList(tasks);
-    updateStorage(tasks);
+    if (event.target.className === 'check') {
+      updateStatus(tasks, parseInt(event.target.parentElement.parentElement.id, 10), true);
+      showList(tasks);
+      updateStorage(tasks);
+    }
+    if (event.target.className === 'fas fa-check') {
+      updateStatus(tasks, parseInt(event.target.parentElement.parentElement.id, 10), false);
+      showList(tasks);
+      updateStorage(tasks);
+    }
+    if (event.target.className === 'fas fa-trash') {
+      tasks = removeTask(tasks, parseInt(event.target.parentElement.id, 10));
+    }
+    if (event.target.className === 'description') {
+      editTask(tasks, event.target);
+      // updateStorage(tasks);
+      // showList(tasks);
+    }
   }
   event.stopPropagation();
 });
 
-listDiv.addEventListener('click', (event) => {
-  if (event.target !== event.currentTarget && event.target.className === 'fas fa-check') {
-    updateStatus(tasks, parseInt(event.target.parentElement.parentElement.id, 10), false);
-    showList(tasks);
-    updateStorage(tasks);
-  }
-  event.stopPropagation();
+clearAllButton.addEventListener('click', () => {
+  tasks = removeCompletedTasks(tasks);
+});
```

Comment:
- [ ] ... and the property `index` set to the value of the new array length (i.e. if a 5th task is added to the list, the index of that task should equal to 5).
- [ ] Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique.